client/my-sites/themes/current-theme/index.jsx

```diff
@@ -56,31 +62,54 @@ const CurrentTheme = React.createClass( {
 				</div>
 				<div className={ classNames(
 					'current-theme__actions',
-					{ 'two-buttons': ! displaySupportButton }
+					{ 'two-buttons': Object.keys( this.props.options ).length === 2 }
 					) } >
-					<CurrentThemeButton name="customize"
-						label={ this.translate( 'Customize' ) }
-						noticon="paintbrush"
-						href={ this.props.canCustomize ? getCustomizeUrl( null, site ) : null }
-						onClick={ this.trackClick } />
-					<CurrentThemeButton name="info"
-						label={ this.translate( 'Info' ) }
-						noticon="info"
-						href={ currentTheme ? getDetailsUrl( currentTheme, site ) : null }
-						onClick={ this.trackClick } />
-					{ displaySupportButton && <CurrentThemeButton name="support"
-						label={ this.translate( 'Setup' ) }
-						noticon="help"
-						href={ currentTheme ? getSupportUrl( currentTheme, site ) : null }
-						onClick={ this.trackClick } /> }
+					{ map( this.props.options, ( option, name ) => (
+						<CurrentThemeButton name={ name }
+							label={ option.label }
+							icon={ option.icon }
+							href={ option.getUrl && currentTheme && option.getUrl( currentTheme ) }
+							onClick={ this.trackClick } />
+					) ) }
 				</div>
 			</Card>
 		);
 	}
 } );
 
+const mergeProps = ( stateProps, dispatchProps, ownProps ) => {
+	const { site } = ownProps;
+	// FIXME (ockham): Remove thus ugly hack. Currently requried since the endpoint doesn't return an `active` attr
+	const theme = Object.assign( {}, stateProps.currentTheme, { active: true } );
```

Comment:
> Setting active would work kinda easy, but unsetting for the previous theme is more of a meh.

I don't think you need to unset it, because the current-theme store should only contain one theme per site: the current one.

But fine to leave it anyway.